From c8207e1cca3f853ca249cc9532674e37906f94cf Mon Sep 17 00:00:00 2001
From: Avula Sri Charan <quic_asrichar@quicinc.com>
Date: Wed, 21 Sep 2022 14:08:24 +0530
Subject: [PATCH] Changes needed for yocto

Currently in OpenWRT we are defining NLA_PUT_S32 which is
not part of OpenWRT's native libnl-tiny package. but as in
yocto where libnl-tiny is not available we used the libnl
which already have the macro defined NLA_PUT_S32.

Hence we are adding an ifndef to avoid compilation error in yocto.

The other change is to avoid the warning declaration-after-statement.

Signed-off-by: Avula Sri Charan <quic_asrichar@quicinc.com>
---
 sar.c  | 3 ++-
 util.c | 4 ++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/sar.c b/sar.c
index 95e36e4..e7f8a9f 100644
--- a/sar.c
+++ b/sar.c
@@ -14,9 +14,10 @@
  * @arg attrtype	Attribute type.
  * @arg value		Numeric value.
  */
+#ifndef NLA_PUT_S32
 #define NLA_PUT_S32(msg, attrtype, value) \
 	NLA_PUT_TYPE(msg, uint16_t, attrtype, value)
-
+#endif
 static int set_sar_specs(struct nl80211_state *state,
 			 struct nl_msg *msg,
 			 int argc, char **argv,
diff --git a/util.c b/util.c
index 48735ab..036f162 100644
--- a/util.c
+++ b/util.c
@@ -1210,7 +1210,8 @@ static void __print_eht_capa(const __u16 *mac_cap,
 {
 	int i;
 	const char *pre = indent ? "\t" : "";
-
+	char *bw[] = { "<= 80", "160", "360" };
+	
 	#define PRINT_EHT_CAP(_var, _idx, _bit, _str) \
 	do { \
 		if (_var[_idx] & BIT(_bit)) \
@@ -1288,7 +1289,6 @@ static void __print_eht_capa(const __u16 *mac_cap,
 	PRINT_EHT_PHY_CAP(3, 14, "MU Beamformer <= 320Mhz");
 	PRINT_EHT_PHY_CAP(3, 15, "TB Sounding Feedback Rate Limit");
 
-	char *bw[] = { "<= 80", "160", "360" };
 	for (i = 0; i < 3; i++) {
 		int j;
 
-- 
2.17.1

